mlflow/pyfunc/model.py

```diff
@@ -1017,12 +1055,14 @@ def _load_context_model_and_signature(
                 python_model_cloudpickle_version,
                 cloudpickle.__version__,
             )
+        python_model_compression = pyfunc_config.get(CONFIG_KEY_COMPRESSION, None)
```

Comment:
This will require that the user sets the compression value correctly when loading a compressed file in an arbitrary compression format. 
Should this be inspecting the stored object filetype extension to ensure that it can load any compressed format? 

For instance, this becomes. a problem for:

User A likes bzip2. Stores all of their models in this format. 
User B likes gzip. 
In User A session, User A needs to load a model that User B stored. If they have their config set to load bzip2, they'll get exceptions raised when loading User B models. 

The config should apply to serialization, but deserialization should detect the compression type and automatically map to the right loader.

Reply:
This uses the fact that `pyfunc_config` is stored with the model so regardless of what User A set in their env variable if they try to load a model created by User B they will decompress using the compression decided by User B.

I added a screenshot to show an example

![Screenshot from 2025-03-25 18-20-46](https://github.com/user-attachments/assets/b07853ea-ed4b-40fa-9291-d7637203a6d3)